### PR TITLE
chore: Updated change log and readme

### PR DESCRIPTION
**Pull request by mnoman09**
## Summary
- Added event processor changes in change log and updated readme

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
+# Optimizely Csharp SDK Changelog
+
+## 3.2.0
```
Comment (mikeng13) [inline CHANGELOG.md]: Version should be 3.3.3 and the date August 22 (aiming to release tomorrow)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+August 19th, 2019
+
+### New Features:
+- Introduced `UserEventFactory` which will return impression and conversion event.
```
Comment (mikeng13) [inline CHANGELOG.md]: Should be in sync with the release notes from the JS SDK: https://github.com/optimizely/javascript-sdk/pull/354

File: README.md [hunk 3]
```diff
@@ -68,7 +68,7 @@ The Optimizely client object accepts the following plug-ins:
 2. `ILogger` exposes a single method, Log, to record activity in the SDK.  An example of a class to bridge the SDK's Log to Log4Net is provided in the Demo Application.
 3. `IErrorHandler` allows you to implement custom logic when Exceptions are thrown.  Note that Exception information is already included in the Log.
 4. `ProjectConfigManager` exposes method for retrieving ProjectConfig instance. Examples include `FallbackProjectConfigManager` and `HttpProjectConfigManager`.
-
+5. `EventProcessor` provide an intermediary processing stage within event production. It's assumed that the EventProcessor dispatches events via a provided EventDispatcher. Examples include `ForwardingEventProcessor` and `BatchEventProcessor`.
```
Comment (mikeng13) [inline README.md]: Nit: provides an intermediary...

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+- Added `EventFactory` taking `UserEvents` and creating Logevent.
+- Introduced `EventProcessor` interface with `BatchEventProcessor` and `ForwardingEventProcessor` implementation.
+- Introduced `LogEvent` notification.
+- Added `BatchEventProcessor` as the default implementation within the `OptimizelyFactory` class.
+
+### Deprecated
+- `EventBuilder` was deprecated and now we will be using `UserEventFactory` and `EventFactory` to create logEvents.
+- `LogEvent` was deprecated from `TrackNotification` and `ActivateNotification` notifications in favor of explicit `LogEvent` notification.
+- New features will no longer be supported on `.net framework 1.6`
```
Comment (mikeng13) [inline CHANGELOG.md]: Is this synonymous with .net core 1.6? Should we can it that instead?

File: CHANGELOG.md (hunk 1 above)
Comment (mjc1283) [reply CHANGELOG.md]: Should be 3.3.0, right?

File: CHANGELOG.md (hunk 2 above)
Comment (mjc1283) [reply CHANGELOG.md]: We could align on the style of these changelogs. JS's is verbose and reads like a summary/guide, but Java and this one are spartan.

File: CHANGELOG.md (hunk 1 above)
Comment (msohailhussain) [reply CHANGELOG.md]: yeah 3.3.0.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,21 @@
...
+
+## 3.3.0
+August 22nd, 2019
+
+### New Features:
+- Added support for event batching via the event processor.
+- Events generated by methods like `activate`, `track`, and `isFeatureEnabled` will be held in a queue until the configured batch size is reached, or the configured flush interval has elapsed. Then, they will be combined into a request and sent to the event dispatcher.
+- To configure event batching, set the `eventBatchSize` and `eventFlushInterval` properties in the `OptimizelyFactory` using `OptimizelyFactory.SetBatchSize(int batchSize)` and `OptimizelyFactory.SetFlushInterval(TimeSpan flushInterval)` and then create `OptimizelyFactory.NewDefaultInstance`.
+- Event batching is enabled by default. `eventBatchSize` defaults to `10`. `eventFlushInterval` defaults to `30000` Milliseconds.
+- Updated the `dispose` method representing the process of closing the instance. When `dispose` is called, any events waiting to be sent as part of a batched event request will be immediately batched and sent to the event dispatcher.
```
Comment (msohailhussain) [inline CHANGELOG.md]: `will be immediately batched or timed out depending on the timed out interval` @mikeng13 @mjc1283 is this suggestion look OK?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,21 @@
...
+- If any such requests were sent to the event dispatcher, `close` returns a `Promise` that fulfills after the event dispatcher calls the response callback for each request. Otherwise, `close` returns an immediately-fulfilled `Promise`.
```
Comment (msohailhussain) [inline CHANGELOG.md]: looks like this is more relevant to javascript. we don't return any promise. Dispose is called that's it.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,21 @@
...
+
+### Deprecated
+- `EventBuilder` was deprecated and now we will be using `UserEventFactory` and `EventFactory` to create logEvents.
```
Comment (msohailhussain) [inline CHANGELOG.md]: have we added warning for eventbuilder ?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,21 @@
...
+- `LogEvent` was deprecated from `TrackNotification` and `ActivateNotification` notifications in favor of explicit `LogEvent` notification.
+- New features will no longer be supported on `.net standard 1.6`
```
Comment (msohailhussain) [inline CHANGELOG.md]: Please add we have deprecated `.net standard 1.6` and `.net 3.5`.

**Comment by msohailhussain** [comment]
as soon as https://github.com/optimizely/csharp-sdk/pull/195 is merged, please rebase with master.

File: CHANGELOG.md (hunk 5 above)
Comment (mjc1283) [reply CHANGELOG.md]: Maybe we can just say `immediately batched and sent to event dispatcher`, then describe the timeout functionality in another bullet point.

**Comment by mikeng13** [comment]
Ping on changes here

**Comment by mikeng13** [comment]
We've decided to postpone the release until Sept 16th. Let's move it there and also include the app config support.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,21 @@
+# Optimizely Csharp SDK Changelog
+
+## 3.3.0
+September 3rd, 2019
+
+### New Features:
+- Added support for event batching via the event processor.
+- Events generated by methods like `activate`, `track`, and `isFeatureEnabled` will be held in a queue until the configured batch size is reached, or the configured flush interval has elapsed. Then, they will be combined into a request and sent to the event dispatcher.
```
Comment (msohailhussain) [inline CHANGELOG.md]: `Activate`, `Track`, `IsFeatureEnabled` typo.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,21 @@
...
+- To configure event batching, set the `eventBatchSize` and `eventFlushInterval` properties in the `OptimizelyFactory` using `OptimizelyFactory.SetBatchSize(int batchSize)` and `OptimizelyFactory.SetFlushInterval(TimeSpan flushInterval)` and then create `OptimizelyFactory.NewDefaultInstance`.
```
Comment (msohailhussain) [inline CHANGELOG.md]: Please use the correct property names. @mnoman09

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,21 @@
...
+- Event batching is enabled by default. `eventBatchSize` defaults to `10`. `eventFlushInterval` defaults to `30000` Milliseconds.
+- Updated the `dispose` method representing the process of closing the instance. When `dispose` is called, any events waiting to be sent as part of a batched event request will be immediately batched and sent to the event dispatcher.
+- If any such requests were sent to the event dispatcher, `close` waits for provided `TimeoutInterval` before closing, so that events get successfully dispatched.
```
Comment (msohailhussain) [inline CHANGELOG.md]: `Close`?

File: CHANGELOG.md (hunk 7 above)
Comment (msohailhussain) [reply CHANGELOG.md]: LogEvents, referring to class.

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,21 @@
...
+
+### Deprecated
+- `EventBuilder` was deprecated and now we will be using `UserEventFactory` and `EventFactory` to create logEvents.
+- `LogEvent` was deprecated from `TrackNotification` and `ActivateNotification` notifications in favor of explicit `LogEvent` notification.
```
Comment (msohailhussain) [inline CHANGELOG.md]: looks bit confusing. We are talking about notification and above this point, we are talking about class. Need to distinguish clearly between these two points.

**Comment by msohailhussain** [comment]
@mnoman09 Please add config support change log and polling start by default in change log. Also update this branch.
Change the release date to 19th September.

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,21 @@
+# Optimizely Csharp SDK Changelog
+
+## 3.3.0
+September 3rd, 2019
```
Comment (msohailhussain) [inline CHANGELOG.md]: date change to 19th september.

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,23 @@
+# Optimizely Csharp SDK Changelog
+
+## 3.3.0
+September 19th, 2019
+
+### New Features:
+- Configuration manager is set to PollingProjectConfigManager and polling will be started by default. 
+- Added support for event batching via the event processor.
+- Events generated by methods like `Activate`, `Track`, and `IsFeatureEnabled` will be held in a queue until the configured batch size is reached, or the configured flush interval has elapsed. Then, they will be combined into a request and sent to the event dispatcher.
+- To configure event batching, set the `MaxEventBatchSize` and `MaxEventFlushInterval` properties in the `OptimizelyFactory` using `OptimizelyFactory.SetBatchSize(int batchSize)` and `OptimizelyFactory.SetFlushInterval(TimeSpan flushInterval)` and then create `OptimizelyFactory.NewDefaultInstance`.
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: nit. `and then creating using ...`

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,23 @@
...
+- Event batching is enabled by default. `eventBatchSize` defaults to `10`. `eventFlushInterval` defaults to `30000` Milliseconds.
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: nit. milliseconds (small m)

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,23 @@
...
+- Updated the `Dispose` method representing the process of closing the instance. When `Dispose` is called, any events waiting to be sent as part of a batched event request will be immediately batched and sent to the event dispatcher.
+- If any such requests were sent to the event dispatcher, `Stop` waits for provided `TimeoutInterval` before stoping, so that events get successfully dispatched.
+- `OptimizelyFactory` now provides support of setting configuration variables from ***App.config***, User will now be able to provide configuration variables of `HttpConfigManager` and `BatchEventProcessor` in ***App.config***. Steps of usage are provided in [README.md](https://github.com/optimizely/csharp-sdk#using-appconfig-in-optimizelyfactory).
+
+### Deprecated
+- `EventBuilder` was deprecated and now we will be using `UserEventFactory` and `EventFactory` to create LogEvent Object.
+- Deprecated `Activate` and `Track` notifications in favor of explicit `LogEvent` notification.
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: We should only mention `Track` notification listener here as `Activate` was already marked as deprecated previously.

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,3 +1,23 @@
+# Optimizely Csharp SDK Changelog
+
+## 3.3.0
+September 19th, 2019
```
Comment (aliabbasrizvi) [inline CHANGELOG.md]: You will need to update date.

**Pull request merged**